includes/class-wc-gateway-amazon-payments-advanced-subscriptions.php

```diff
@@ -318,27 +318,32 @@ public function recurring_checkout_session_update( $payload, $checkout_session_i
 			'amount'    => null,
 		);
 
+		$recurring_total = 0;
+		foreach ( WC()->cart->recurring_carts as $recurring_cart ) {
+			$recurring_total += $recurring_cart->get_total( 'edit' );
+		}
+
+		$recurring_total = wc_format_decimal( $recurring_total, '' );
+
```

Comment:
taking this out of the conditional won't give a notice on no recurrings carts?

Reply:
No, because we're checking subscriptions in cart above. If there's no subscriptions in cart, then we won't get to this point. We have 2 different checks actually.